Revamp online courses page

adpi2 opened the pull request:
The goal of this PR is to improve the organization and visibility of online courses by:
- unifying content which is spread across [online-courses.html](https://docs.scala-lang.org/online-courses.html) and [learn.html](https://docs.scala-lang.org/learn.html)
- giving greater prominence to Scala Center and Rock the JVM courses 
- reorganizing other online resources to prioritize fresh content

The old/current version of [online-courses.html](https://docs.scala-lang.org/online-courses.html) is linked from the new page, but it will no longer be directly linked from the main page or navbar. While it provides valuable information for organizations, it may not be as relevant for individual learners.

![Screenshot 2024-11-19 at 15-46-00 Online Courses Scala Documentation](https://github.com/user-attachments/assets/0ce97a82-7c76-4c63-baf5-4f2869226ea8)

SethTisue committed:
add JDK 23 to build tool table

SethTisue committed:
as far as we know, sbt 1.9.0 works on JDK 23

works for me on a sample repo, anyway. if we discover otherwise
we can change it

SethTisue committed:
Update _overviews/jdk-compatibility/overview.md

Co-authored-by: Tobias Roeser <le.petit.fou@web.de>

adpi2 committed:
Rename online-courses.html to scalacenter-courses.html

adpi2 committed:
Rename learn.html to online-courses.html

adpi2 committed:
Add details about Coursera and Extension School

adpi2 committed:
Remove Scastie from Online Courses

It's in the main navigation bar already, as Playground.

adpi2 committed:
Move Rock the JVM up

adpi2 committed:
Add Other Online Resources section

adpi2 committed:
Split 'Tour of Scala' and 'Scala Exercices'

Also move 'Scala Learning Community' to 'Tour of Scala'
It's not an official Scala discord and it is maintained by the Tour of Scala community

adpi2 committed:
Reorder other resources

Most up-to-date first

adpi2 commented:
I've just noticed that I removed the list of contributors of the page, but I can add it back if necessary.

gkepka committed:
Consistent terminology on type parameters

sjrd committed:
Merge pull request #3112 from gkepka/type-parameters

Consistent terminology on type parameters

adpi2 committed:
Add new online-courses layout with catchy images

gkepka committed:
Consistent Scala 3 specific page warnings

sjrd committed:
Merge pull request #3113 from gkepka/scala-3-only

Consistent Scala 3 specific page warnings

gkepka committed:
Remove mention of XML processing in the Tour of Scala

sjrd committed:
Merge pull request #3116 from gkepka/scala-tour-xml

Remove mention of XML processing in the Tour of Scala

Friendseeker committed:
Add documentation for Pattern Matching on String

Friendseeker committed:
Change String -> string as other h2 headings use lower case

Friendseeker committed:
Add note on extractor object

Friendseeker committed:
Fix CI via removing dead links (#3119)

Friendseeker committed:
Adopt feedback from adpi2

adpi2 committed:
Re-order navbar links

"Scala 3" is too vague and confusing. To learn Scala 3, one should click on "Learn" instead of "Scala 3".
To learn the differences between Scala 2 and Scala 3, one should click on "Scala 3".

To make things clearer:
- Moved "Learn" before "Scala 3"
- Renamed "Scala 3" to "Scala 3 Migration"

adpi2 committed:
Merge pull request #3121 from adpi2/reorder-links

Re-order navbar links

SethTisue commented:
@adpi2 might you be interested in addressing https://github.com/scala/docs.scala-lang/issues/2946 while you're revamping?

dependabot[bot] committed:
Bump webrick from 1.9.0 to 1.9.1

Bumps [webrick](https://github.com/ruby/webrick) from 1.9.0 to 1.9.1.
- [Release notes](https://github.com/ruby/webrick/releases)
- [Commits](https://github.com/ruby/webrick/compare/v1.9.0...v1.9.1)

---
updated-dependencies:
- dependency-name: webrick
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

SethTisue committed:
modernize using syntax, bump versions

gkepka committed:
Scala 2 Book migration - Prelude and Preliminaries

gkepka committed:
Scala 2 Book migration - Control Structures

gkepka committed:
Add match following a period example and remove languages from Partial Functions page

gkepka committed:
Apply suggestions from review

gkepka committed:
Update _overviews/scala3-book/control-structures.md

Co-authored-by: Sébastien Doeraene <sjrdoeraene@gmail.com>

gkepka committed:
Merge pull request #3126 from gkepka/scala-book-control-structures

Scala 2 Book migration - Control Structures

gkepka committed:
Merge pull request #3125 from SethTisue/remove-extra-quotes

modernize using syntax, bump versions

gkepka committed:
Merge pull request #3123 from scala/dependabot/bundler/webrick-1.9.1

Bump webrick from 1.9.0 to 1.9.1

WojciechMazur committed:
Add Scala 3.6.2 to docs config

WojciechMazur committed:
Update JDK Scala compatibility table

scala-improvement-bot committed:
Update SIPs state

gkepka committed:
Merge pull request #3127 from gkepka/scala-book-prelude-preliminaries

Scala 2 Book migration - Prelude and Preliminaries

SethTisue committed:
add our Bluesky, and change Twitter->X

adpi2 committed:
Merge pull request #3129 from SethTisue/bluesky-and-x

add our Bluesky, and change Twitter->X

ErikPohl444 committed:
Update pattern-matching.md

Added a note about sealed traits because it is the first usage of them in the guide.

SethTisue committed:
profile is verified now

ErikPohl444 committed:
Update pattern-matching.md

Added a link to sealed types below

ErikPohl444 committed:
Update singleton-objects.md

Added a note about 'Option', 'Some', and 'None'

ErikPohl444 committed:
Update _tour/singleton-objects.md

Thank you for the edit.  Accepted.

Co-authored-by: adpi2 <adrien.piquerez@gmail.com>

ErikPohl444 committed:
Update _tour/pattern-matching.md

Thanks for catching this.  I learned a bit more about successful markdown links.  I accept the edit.

Co-authored-by: Gabriel Kępka <44374670+gkepka@users.noreply.github.com>

gkepka committed:
Merge pull request #3130 from ErikPohl444/patch-1

Update pattern-matching.md

adpi2 committed:
Merge pull request #3131 from ErikPohl444/patch-2

Update singleton-objects.md

adpi2 committed:
Merge pull request #3118 from Friendseeker/patch-1

Add documentation for Pattern Matching on `String`

scala-improvement-bot committed:
Update SIPs state

pochka15 committed:
Change the performance of the Queue prepend operation

evis committed:
rm redundant space

Philippus committed:
Merge pull request #3137 from evis/patch-1

rm redundant space

adpi2 committed:
Merge pull request #3135 from pochka15/queue-performance

Change the performance of the Queue prepend operation

Toxapex4096 committed:
Translated Scala Introduction for Python developers to Japanese (#3136)

Translated docs to Japanese

SethTisue committed:
Merge remote-tracking branch 'origin/main' into jdk-23-build-tools

SethTisue committed:
add scala-cli to build tool table

SethTisue committed:
Lightbend -> Akka

SethTisue committed:
Merge pull request #3082 from SethTisue/jdk-23-build-tools

add JDK 23 to build tool table

SethTisue committed:
add JDK 24 to compat page (#3140)

keyno63 committed:
update scalatest version to latest (3.2.19) in japanese

keyno63 committed:
update scalatest version

keyno63 committed:
update scalatest version, in description

adpi2 committed:
Merge pull request #3141 from keyno63/fix/scalatest-version

update scalatest version

SethTisue committed:
Scala 2.13.16 (#3143)

SethTisue committed:
FAQ: add a sentence to stdlib version number question

SethTisue committed:
Toolkit docs: fix 404 link to sttp websockets page

adpi2 committed:
Merge pull request #3145 from SethTisue/fix-dead-sttp-link

Toolkit docs: fix 404 link to sttp websockets page

sjrd committed:
Merge pull request #3144 from SethTisue/faq-tweak-2

FAQ: add a sentence to stdlib version number question

WojciechMazur committed:
Add Scala 3.6.3 API docs links (#3146)

Flowdalic committed:
string-interpolation: fix headline level

Previously, the raw-interpolator was at the third headline
level (###), while the s-interpolator and the f-interpolator where at
the second headline level (##).

This changes and renames the second headline "Other Interpolators" to
become a second level headline, under which we place all three string
interpolators.

Flowdalic committed:
string-interpolation: the raw-interpolator also performs variable substitution

Clarify that the raw-interpolator also performs variable
substitution. This was not obvious from the text and it is easy to
believe that raw means just that: the raw string, as is, without any
substitutions performed.

This also moves the last sentence of the raw-interpolator section
into the following section, where it logicaly belongs.

sjrd committed:
Merge pull request #3147 from Flowdalic/string-interpolators

string-interpolation: fix headline level

sjrd committed:
Merge pull request #3148 from Flowdalic/raw-string-interpolation

string-interpolation: the raw-interpolator also performs variable substitution

scala-improvement-bot committed:
Update SIPs state

WojciechMazur committed:
Add API docs for 3.3.5 LTS and update compatibility table (#3151)

tgodzik committed:
docs: Add link to recent blogpost (#3152)

adpi2 committed:
Expand on Rock the JVM and bring it up

adpi2 commented:
Moved to https://github.com/scala/docs.scala-lang/pull/3155
